tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedTestFormatTestRunner.cs

```diff
@@ -0,0 +1,217 @@
+                runOnRequirements: testCase.Shared.GetValue("runOnRequirements", null)?.AsBsonArray,
+                entities: testCase.Shared["createEntities"].AsBsonArray, // TODO: can be absent
+                initialData: testCase.Shared.GetValue("initialData", null)?.AsBsonArray,
+                test: testCase.Test);
+        }
+
+        public void Run(
+            string schemaVersion,
+            BsonArray runOnRequirements,
+            BsonArray entities,
+            BsonArray initialData,
+            BsonDocument test)
+        {
+            schemaVersion.Should().StartWith("1.0");
+            if (runOnRequirements != null)
+            {
+                RequireServer.Check().RunOn(runOnRequirements);
+            }
+
+            _entityMap = new EntityMap(entities);
+
+            if (initialData != null)
+            {
+                AddInitialData(DriverTestConfiguration.Client, initialData);
+            }
+
+            foreach (var operationItem in test["operations"].AsBsonArray)
+            {
+                var operation = CreateOperation(operationItem.AsBsonDocument, _entityMap);
+                var result = ExecuteOperation(operation, test["async"].AsBoolean);
+                if (result != null)
+                {
+                    AssertResult(operationItem.AsBsonDocument, result, _entityMap);
+                }
+                if (operation is UnifiedFailPointOperation failPointOperation)
+                {
+                    _failPoints.Add(failPointOperation.FailPoint);
+                }
+            }
+
+            if (test.AsBsonDocument.TryGetValue("outcome", out var expectedOutcome))
+            {
+                AssertOutcome(DriverTestConfiguration.Client, expectedOutcome.AsBsonArray);
+            }
+            if (test.AsBsonDocument.TryGetValue("expectEvents", out var expectedEvents))
+            {
+                AssertEvents(expectedEvents.AsBsonArray, _entityMap);
+            }
+        }
+
+        public void Dispose()
+        {
+            foreach (var failPoint in _failPoints)
+            {
+                failPoint?.Dispose();
+            }
+            _entityMap?.Dispose();
+        }
+
+        // private methods
+        private void AddInitialData(IMongoClient client, BsonArray initialData)
+        {
+            foreach (var dataItem in initialData)
+            {
+                var collectionName = dataItem["collectionName"].AsString;
+                var databaseName = dataItem["databaseName"].AsString;
+                var documents = dataItem["documents"].AsBsonArray.Cast<BsonDocument>().ToList();
+
+                var database = client.GetDatabase(databaseName);
+                var collection = database
+                    .GetCollection<BsonDocument>(collectionName)
+                    .WithWriteConcern(WriteConcern.WMajority);
+
+                database.DropCollection(collectionName);
+                if (documents.Any())
```

Comment:
It's a  minor point but the following is more efficient:

                if (documents.Count > 0)

It's because `Any` is a LINQ method and involves creating an enumerator, whereas `Count` is a direct property on `List<T>`.